code/modules/admin/verbs/randomverbs.dm

```diff
@@ -1190,3 +1190,57 @@ Traitors and the like can also be revived with the previous role mostly intact.
 					if(!source)
 						return
 			REMOVE_TRAIT(D, chosen_trait, source)
+
+/client/proc/create_crate(object as text)
+	set name = "Create Crate"
+	set desc = "Spawn a crate from a supplypack datum. Append a period to the text in order to exclude subtypes of paths matching the input."
+	set category = "Event"
+
+	if(!check_rights(R_SPAWN))
+		return
+
+	var/list/types = list()
+	for(var/typepath in subtypesof(/datum/supply_packs))
+		var/datum/supply_packs/P = typepath
+		if(initial(P.name) == "HEADER")
+			continue // To filter out group headers
+		types += P
```

Comment:
```suggestion
	for(var/typepath in SSeconomy.supply_packs)
		types += typepath
```
This might work? The paths are stored as strings in the SS list and if it works you can avoid having to filter it yourself

Reply:
My one concern with this is if the SSeconomy subsystem fails and cant recover (or loses the supply_packs var), admins may not be able to spawn crates (which is what this tool is supposed to help with if the cargo/vendor system fails). Thoughts?

Reply:
Pretty sure that wouldn't break it. Even if SSeconomy fails and can't fire, its vars are still accessible.